src/RecipeBot/Controllers/RecipeController.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (C) 2022 Dennis Tang. All rights reserved.
+//
+// This file is part of RecipeBot.
+//
+// RecipeBot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+using System.Threading.Tasks;
+using Common.Utils;
+using Discord;
+using Discord.Common;
+using RecipeBot.Discord.Controllers;
+using RecipeBot.Discord.Data;
+using RecipeBot.Discord.Views;
+using RecipeBot.Domain.Exceptions;
+using RecipeBot.Domain.Factories;
+using RecipeBot.Domain.Models;
+using RecipeBot.Exceptions;
+using RecipeBot.Services;
+
+namespace RecipeBot.Controllers;
+
+/// <summary>
+/// A concrete implementation of the <see cref="IRecipeController"/>.
+/// </summary>
+public class RecipeController : IRecipeController
+{
+    private readonly ILoggingService logger;
+    private readonly RecipeModelCreationService modelCreationService;
+
+    public RecipeController(IRecipeModelCharacterLimitProvider limitProvider,
```

Comment:
Limit provider should be moved from the RecipeBot.Discord to the RecipeBot assembly